### PR TITLE
Adds maximum temp values to rock strata

### PR DESCRIPTION
**Pull request by comma**
## Description of changes
They were previously set to default infinite value

Added values from cursory google search

## Why and what will this PR improve
It resulted in volcanic caves etc generating with sandstone rocks and such, immersion shattered

## Authorship

Me

## Changelog
Doubt anyone notices

### CODE REVIEW

File: code/modules/materials/geology/strata_metamorphic.dm [hunk 1]
```diff
@@ -2,6 +2,7 @@
 	name = "metamorphic rock"
 	base_materials = list(/decl/material/solid/stone/marble)
 	default_strata_candidate = TRUE
+	maximum_temperature = T0C + 600
```
Comment (MistakeNot4892) [inline code/modules/materials/geology/strata_metamorphic.dm]: Shouldn't this just use the melting point of the base materials?

Comment (MistakeNot4892) [reply code/modules/materials/geology/strata_metamorphic.dm]: Per Discord, strata use the melting point of their base materials but it's not set properly on marble/etc.

Comment (comma) [reply code/modules/materials/geology/strata_metamorphic.dm]: Done

File: code/modules/materials/definitions/solids/materials_solid_stone.dm [hunk 2]
```diff
@@ -66,6 +67,7 @@
 	wall_support_value = MAT_VALUE_VERY_HEAVY
 	hardness = MAT_VALUE_HARD
 	reflectiveness = MAT_VALUE_SHINY
+	melting_point  = T0C + 600
```
Comment (out-of-phaze) [inline code/modules/materials/definitions/solids/materials_solid_stone.dm]: > The melting temperature of dry granite at ambient pressure is 1215–1260 °C (2219–2300 °F); it is strongly reduced in the presence of water, down to 650 °C at a few hundred megapascals of pressure.

Are we sure 600C is high enough?

**Review by MistakeNot4892 — DISMISSED** [review]

File: code/modules/materials/definitions/solids/materials_solid_stone.dm [hunk 3]
```diff
@@ -32,6 +32,7 @@
 	uid = "solid_sandstone"
 	lore_text = "A clastic sedimentary rock. The cost of boosting it to orbit is almost universally much higher than the actual value of the material."
 	value = 1.5
+	melting_point = T0C + 400
```
Comment (out-of-phaze) [inline code/modules/materials/definitions/solids/materials_solid_stone.dm]: It seems that silicon dioxide (quartz, the main component of sandstone) melts at 1,713C? Which is a bit high. It looks like heating it to 600C might cause it to start fracturing so we could probably use that as a limit.

**Review by out-of-phaze — CHANGES_REQUESTED** [review]

File: code/modules/materials/definitions/solids/materials_solid_stone.dm (hunk 2 above)
Comment (comma) [reply code/modules/materials/definitions/solids/materials_solid_stone.dm]: Made it 1200

File: code/modules/materials/definitions/solids/materials_solid_stone.dm (hunk 3 above)
Comment (comma) [reply code/modules/materials/definitions/solids/materials_solid_stone.dm]: Made it 600

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**